package/boot/uboot-mediatek/patches/107-configs-add-useful-options-to-mt7981-rfb.patch

```diff
@@ -1,472 +1,452 @@
-+CONFIG_CMD_FS_UUID=y
  CONFIG_CMD_GPIO=y
  CONFIG_CMD_MTD=y
-+CONFIG_CMD_HASH=y
-+CONFIG_CMD_ITEST=y
-+CONFIG_CMD_LED=y
-+CONFIG_CMD_LICENSE=y
-+CONFIG_CMD_LINK_LOCAL=y
-+CONFIG_CMD_MTD=y
-+CONFIG_CMD_NAND=y
-+# CONFIG_MTD_RAW_NAND is not set
-+CONFIG_CMD_NAND_TRIMFFS=y
+-CONFIG_CMD_PING=y
 +CONFIG_CMD_PCI=y
++CONFIG_CMD_TFTPSRV=y
++CONFIG_CMD_RARP=y
++CONFIG_CMD_CDP=y
++CONFIG_CMD_SNTP=y
++CONFIG_CMD_LINK_LOCAL=y
++CONFIG_CMD_DNS=y
++CONFIG_CMD_CACHE=y
 +CONFIG_CMD_PSTORE=y
 +CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
-+CONFIG_CMD_SF=y
-+CONFIG_CMD_SF_TEST=y
- CONFIG_CMD_PING=y
-+CONFIG_CMD_PXE=y
++CONFIG_CMD_UUID=y
++CONFIG_CMD_HASH=y
  CONFIG_CMD_SMC=y
++CONFIG_CMD_FS_UUID=y
  CONFIG_CMD_UBI=y
  CONFIG_CMD_UBI_RENAME=y
-+CONFIG_CMD_UBIFS=y
-+CONFIG_CMD_TFTPBOOT=y
-+CONFIG_CMD_TFTPSRV=y
-+CONFIG_CMD_ASKENV=y
-+CONFIG_CMD_RARP=y
-+CONFIG_CMD_SETEXPR=y
-+CONFIG_CMD_SLEEP=y
-+CONFIG_CMD_SNTP=y
-+CONFIG_CMD_SOURCE=y
-+CONFIG_CMD_STRINGS=y
-+CONFIG_CMD_UUID=y
  CONFIG_ENV_OVERWRITE=y
 +CONFIG_ENV_IS_IN_UBI=y
++CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 +CONFIG_ENV_UBI_PART="ubi"
-+CONFIG_ENV_SIZE=0x1f000
-+CONFIG_ENV_SIZE_REDUND=0x1f000
 +CONFIG_ENV_UBI_VOLUME="ubootenv"
 +CONFIG_ENV_UBI_VOLUME_REDUND="ubootenv2"
++CONFIG_SYS_RELOC_GD_ENV_ADDR=y
  CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
  CONFIG_NET_RANDOM_ETHADDR=y
-+CONFIG_REGEX=y
  CONFIG_REGMAP=y
  CONFIG_SYSCON=y
++CONFIG_BUTTON=y
++CONFIG_BUTTON_GPIO=y
  CONFIG_CLK=y
++CONFIG_GPIO_HOG=y
++CONFIG_LED=y
++CONFIG_LED_BLINK=y
++CONFIG_LED_GPIO=y
  # CONFIG_MMC is not set
-+CONFIG_HUSH_PARSER=y
-+CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
-+CONFIG_SYS_RELOC_GD_ENV_ADDR=y
  CONFIG_MTD=y
  CONFIG_DM_MTD=y
- CONFIG_MTK_SPI_NAND=y
+@@ -50,6 +86,6 @@ CONFIG_PINCTRL_MT7981=y
+ CONFIG_POWER_DOMAIN=y
+ CONFIG_MTK_POWER_DOMAIN=y
+ CONFIG_DM_SERIAL=y
++CONFIG_SERIAL_RX_BUFFER=y
+ CONFIG_MTK_SERIAL=y
+ CONFIG_HEXDUMP=y
+-CONFIG_LMB_MAX_REGIONS=64
 --- a/configs/mt7981_nor_rfb_defconfig
 +++ b/configs/mt7981_nor_rfb_defconfig
-@@ -12,7 +12,22 @@ CONFIG_DEBUG_UART_BASE=0x11002000
+@@ -5,37 +5,74 @@ CONFIG_ARCH_MEDIATEK=y
+ CONFIG_TEXT_BASE=0x41e00000
+ CONFIG_SYS_MALLOC_F_LEN=0x4000
+ CONFIG_NR_DRAM_BANKS=1
++CONFIG_ENV_SIZE=0x4000
++CONFIG_ENV_OFFSET=0x0
+ CONFIG_DEFAULT_DEVICE_TREE="mt7981-rfb"
+-CONFIG_SYS_PROMPT="MT7981> "
++CONFIG_OF_LIBFDT_OVERLAY=y
+ CONFIG_TARGET_MT7981=y
++CONFIG_SYS_LOAD_ADDR=0x46000000
+ CONFIG_DEBUG_UART_BASE=0x11002000
  CONFIG_DEBUG_UART_CLOCK=40000000
- CONFIG_SYS_LOAD_ADDR=0x46000000
+-CONFIG_SYS_LOAD_ADDR=0x46000000
  CONFIG_DEBUG_UART=y
 -# CONFIG_AUTOBOOT is not set
-+CONFIG_OF_LIBFDT_OVERLAY=y
-+CONFIG_SMBIOS_PRODUCT_NAME=""
-+CONFIG_CFB_CONSOLE_ANSI=y
-+CONFIG_BOARD_LATE_INIT=y
-+CONFIG_BUTTON=y
-+CONFIG_BUTTON_GPIO=y
-+CONFIG_GPIO_HOG=y
-+CONFIG_CMD_ENV_FLAGS=y
 +CONFIG_FIT=y
-+CONFIG_FIT_ENABLE_SHA256_SUPPORT=y
-+CONFIG_LED=y
-+CONFIG_LED_BLINK=y
-+CONFIG_LED_GPIO=y
-+CONFIG_SPI_BOOT=y
-+CONFIG_BOOTSTD_DEFAULTS=y
 +CONFIG_BOOTSTD_FULL=y
++CONFIG_SPI_BOOT=y
++CONFIG_OF_SYSTEM_SETUP=y
```

Comment:
> While I get most changes, the 341 patch drop and RFB config I dont understand

@robimarko The MT7981 RFB config symbol `CONFIG_OF_SYSTEM_SETUP=y` in patch 341 has been merged into this patch to simplify the config refreshing work. These crazy diff changes are caused by an auto refresh `make  savedefconfig`. The default .config files are same as before.

Reply:
Please don't do that. Adding `CONFIG_OF_SYSTEM_SETUP` belongs into the patch adding support for setting the root device in DT -- and that patch may change as there is an upstream debate about implementing that feature in vanilla U-Boot. If at all, then adding the config symbol should be moved out of that patch for all 4 SoCs (MT7622, MT7981, MT7986 and MT7988).

Reply:
I saw that mt7988-rfb did a similar thing, so I merged it. Should we add another commit to split them?

ref: https://git.openwrt.org/?p=openwrt%2Fopenwrt.git&a=search&h=HEAD&st=grep&s=CONFIG_OF_SYSTEM_SETUP%3Dy